src/OpenTelemetry.Api/Trace/TelemetrySpan.cs

```diff
@@ -64,6 +64,24 @@ public bool IsRecording
             }
         }
 
+        /// <summary>
+        /// Gets the identity of the parent span id, if any.
+        /// </summary>
+        public ActivitySpanId ParentId
```

Comment:
Would it be better to expose parentspancontext instead? (consisting of all fields from SpanContext)

Reply:
I think so and I tried to do that first, but the `ActivityContext` doesn't include the `ParentId` - that's incorporated onto the `Activity` type itself, so I'd need to change the `SpanContext` constructor to pass that information along as a separate field. I wanted to keep the changes to a bare minimum until I'd spoken with a project member but I'd be happy to re-arrange that.

Reply:
Yes, SpanContext (ActivityContext) does not include ParentId, and is as per spec, so we don't need to change it.
The spec says Span should have a way to get parent span either in the form of a Span itself, or `SpanContext`.

Activity currently exposes `Parent` (for in-proc parent), and `ParentId` and `ParentSpanId` for out-of-proc parent. The `ParentId` is in the W3CTraceParent format, and we may have to parse it to convert to `Activity/SpanContext`.

If your requirement is met by simply having `ParentSpanId` (as in this PR), lets start with that now, and expose more later. If this is the case, then rename the field to `ParentSpanId` to make it clears its the SpanId only.

Reply:
```suggestion
        public ActivitySpanId ParentSpanId
```

Reply:
+1 on adding a `ParentSpanContext` to better reflect the spec.

I think you can construct the `SpanContext` for the parent from a `TelemetrySpan` like so:

```csharp
SpanContext? parent = this.Activity.Parent != null
    ? new SpanContext(this.Activity.Parent.Context)
    : null;
```

Reply:
Ok, so is that what we should do? 'ParentContext'?

Reply:
@alanwest `this.Activity.Parent` would be null, if the parent is not an in-proc one. In that case, `Activity.ParentId` could be used to construct the SpanContext, but i am not sure if thats fully correct.

Reply:
Ah yes that's right. Sorry 😅 you said this, but I didn't quite follow at first, thanks for the clarification. Agreed, makes sense to move forward with a `ParentSpanId` property.